Add RunJS plugin

eoureo opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["description": "RunJS is a plugin for running JavaScript code in Obsidian (https://obsidian.md).",](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/manifest.json#L6)
We have recently published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.
Avoid including sentences like "This is a plugin for Obsidian" in your description.

[import { posix as path } from "path";](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/src/main.ts#L14)
These imports are only available from a NodeJS runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[id: "runjs-open-modal",](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/src/main.ts#L204)
Obsidian will automatically prefix the command id with the plugin id internally,
you don't need to include it yourself.

[name: "Open RunJS Code modal",](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/src/main.ts#L205)
I would consider changing this name, this will show up as `RunJS: Open RunJS code modal`

[const tFolder: TFolder = <TFolder>(](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/src/main.ts#L556-L558)
Before using check if it is actually a TFolder with a `instanceOf` check, `getAbstractFileByPath()` can also return files.

[get iconsObsidian(): string[] {](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/src/main.ts#L1080-L1098)
Remove this code, and use the `getIconIds` function instead.


[.addButton((bt) => {](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/src/settingtab.ts#L302-L308)
[Avoid using `outerHTML`](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%60innerHTML%60%2C+%60outerHTML%60+and+%60insertAdjacentHTML%60)

[sleep(50);](https://github.com/eoureo/obsidian-runjs/blob/80f192ec4bb5754133dad56a69734ad3f2dad5e3/src/codelist_view.ts#L639-L640)
Why is this important?

eoureo commented:
Thanks for your comments.
I have updated the code as you suggested.

New version released: https://github.com/eoureo/obsidian-runjs/releases/tag/0.4.0


- changed plugin descriptions.
- deleted `import { posix as path } from "path";`.
- changed command id to "open-codelist-modal".
- changed command name to "Open codelist modal",
- changed to check tFolder with a instanceOf check,
- changed button not using outerHTML.
- deleted `sleep(50);`.
